### ISSUE TITLE
Training fails after a few steps

### ISSUE THREAD

**Issue by yuvalvarkel** [ShannonAI/CorefQA#14]
Training fails after ~10 steps. From the stacktrace it appears the underlying BERT/SpanBERT receives an input sequence that is too long.
I added a tf.Print on the shape of qa_input_mask in corefqa_model.py and saw it is 606 in one of the steps (>512).
The failed assert is in bert/modeling.py, line 495, in embedding_postprocessor
    `assert_op = tf.assert_less_equal(seq_length, max_position_embeddings)`

btw, I assume that the empty folder named bert in this repo should be filled with the content of https://github.com/google-research/bert

Please advise.

**Comment by stepperL** [comment]
Yeah, this is because of the memory issue. You need TPU to get the model trained.

**Comment by yuvalvarkel** [comment]
> Yeah, this is because of the memory issue. You need TPU to get the model trained.

@stepperL  To replicate this bug I just trained on CPU with 512GB of ram available, so this one is not caused by memory exhaustion. Obviously training the model fully on CPU won't be time-efficient, I did that just to make sure the code doesn't have other issues, and it does.

When trying to train on GPU or TPU I have other issues as I commented in a different issue about memory, copying here also:

what TPU configuration did you use? Did you use the same code version as published here?

I tried running it on a TPU and it fails since the file_based_input_fn_builder returns features with dynamic shape, which are not supported on TPU. Am I missing something?

I also tried to train on V100 GPU with 32GB, batch size = 1 and it fails on memory exhaustion.
Any advice on how to train the model?

**Comment by lu161513** [comment]
> > Yeah, this is because of the memory issue. You need TPU to get the model trained.
> 
> @stepperL To replicate this bug I just trained on CPU with 512GB of ram available, so this one is not caused by memory exhaustion. Obviously training the model fully on CPU won't be time-efficient, I did that just to make sure the code doesn't have other issues, and it does.
> 
> When trying to train on GPU or TPU I have other issues as I commented in a different issue about memory, copying here also:
> 
> what TPU configuration did you use? Did you use the same code version as published here?
> 
> I tried running it on a TPU and it fails since the file_based_input_fn_builder returns features with dynamic shape, which are not supported on TPU. Am I missing something?
> 
> I also tried to train on V100 GPU with 32GB, batch size = 1 and it fails on memory exhaustion.
> Any advice on how to train the model?

I have the same question. I use Tesla P40 GPU with 24GB, batch size =1 and it fails on memory exhaustion. Have you solved the problem？

**Comment by stepperL** [comment]
I remember that Wei. conducted experiments on a Cloud TPU v3-8 core (may correspond to a GPU with 64GB memory).
I am rushing for deadlines currently. But I do plan to clarify the questions starting from Jun 2.  
Many Thanks!
